langstream-pulsar-runtime/src/main/java/ai/langstream/pulsar/runner/PulsarTopicConnectionsRuntimeProvider.java

```diff
@@ -524,19 +532,51 @@ private class PulsarTopicProducer<K> implements TopicProducer {
             Producer<K> producer;
             Schema<K> schema;
 
+            static final Map<Class<?>, Schema<?>> BASE_SCHEMAS =
+                Map.ofEntries(
+                    entry(String.class, Schema.STRING),
+                    entry(Boolean.class, Schema.BOOL),
+                    entry(Byte.class, Schema.INT8),
+                    entry(Short.class, Schema.INT16),
+                    entry(Integer.class, Schema.INT32),
+                    entry(Long.class, Schema.INT64),
+                    entry(Float.class, Schema.FLOAT),
+                    entry(Double.class, Schema.DOUBLE),
+                    entry(byte[].class, Schema.BYTES),
+                    entry(Date.class, Schema.DATE),
+                    entry(Timestamp.class, Schema.TIMESTAMP),
+                    entry(Time.class, Schema.TIME),
+                    entry(LocalDate.class, Schema.LOCAL_DATE),
+                    entry(LocalTime.class, Schema.LOCAL_TIME),
+                    entry(LocalDateTime.class, Schema.LOCAL_DATE_TIME),
+                    entry(Instant.class, Schema.INSTANT),
+                    entry(ByteBuffer.class, Schema.BYTEBUFFER)
+                );
+
+
             public PulsarTopicProducer(Map<String, Object> configuration) {
                 this.configuration = configuration;
             }
 
             @Override
             @SneakyThrows
             public void start() {
-                String topic = (String) configuration.remove("topic");
-                schema = (Schema<K>) configuration.remove("schema");
-                if (schema == null) {
-                    schema = (Schema) Schema.STRING;
+                if (configuration.containsKey("valueSchema")) {
+                    SchemaDefinition valueSchemaDefinition =
+                            mapper.convertValue(
+                                    configuration.remove("valueSchema"), SchemaDefinition.class);
```

Comment:
is is safe to "remove" ?

Reply:
we work on a copy.
It's also because we call producer.load(configuration) later.

Reply:
thanks for your clarification